update/Pos/Update/Poll/PollState.hs

```diff
@@ -57,7 +58,9 @@ data PollState = PollState
       -- | Update proposals for each application
     , _psActivePropsIdx     :: !(HM.HashMap ApplicationName (HashSet UpId))
       -- | Slotting data for this node
-    , _psSlottingData       :: !SlottingData
+    , _psEpochSlottingData  :: !(HashMap EpochIndex EpochSlottingData)
+      -- | Last updated epoch index
+    , _psEpochLastIndex  :: !EpochIndex
```

Comment:
Hm, do you need to store it explicitly? I think in this case it makes sense to store slotting data in `Map` (from `containers`) rather than `HashMap`, in this case you can get last index using `findMax` function.
Update: while I was writing it I realized that your approach also makes sense and it's subjective. I think `findMax` approach makes code simpler, but `HashMap + EpochIndex` is slightly more efficient. So I don't mind if you leave it as is.

P. S. I started reviewing from bottom for some reason :) So don't be surprised to this comment _here_ (probably it can be applied to other places as well).